packages/engine/lib/execution/src/runner/comms/mod.rs

```diff
@@ -115,11 +119,28 @@ pub struct NewSimulationRun {
 }
 
 #[derive(Clone)]
+#[non_exhaustive]
 pub struct DatastoreSimulationPayload {
     pub agent_batch_schema: Arc<AgentSchema>,
     pub message_batch_schema: Arc<Schema>,
     pub context_batch_schema: Arc<Schema>,
-    pub shared_store: Arc<SharedStore>,
+    pub shared_store: Weak<SharedStore>,
```

Comment:
Is there anywhere else we might want to swap `Arc`s over to `Weak`? 

Is there a reason not to in most cases where we have a similar pattern of something sort of being owned by one of the controllers?